### ISSUE TITLE
Function/setting to set edge class

### ISSUE THREAD

**Issue by umdstu** [cytoscape/cytoscape.js-edgehandles#138]
Feature Request (I think):

For maximum customizability, I use classes to style all of my nodes and edges. With this extension, the edges created are all straight lines unless you define the styles in the cytoscape configuration params.

I tried to set the class in the edgeParams(s,t,i) callback, but realized it doesn't touch classes, they are hard-coded internally. Would it be possible to allow the classes to be overridden as well via an `options.edgeClasses()` callback function/configuration?

**Comment by umdstu** [comment]
After some more time to play with this I think a better way forward would be a data() function as an optional config argument that gets merged in with the in-line data (source/target).

```javascript
// flat
var source2target = cy.add(getEleJson({
  group: 'edges',
  data: Object.assign({
    source: source.id(),
    target: target.id()
  }, options.data())
}, options.edgeParams(source, target, 0), classes));
```

**Comment by umdstu** [comment]
Nevermind, feel free to close this. Passing `data` key in the options.edgeParams that already exists seems to have the same effect.

**Comment by maxkfranz** [comment]
Do you experience this issue only for preview elements?  Are the finally added edges also affected?

**Comment by umdstu** [comment]
It was for both. Using `edgeParams()` and passing `data: {}` i was able to set the added edges style. Additionally, using `ghostEdgeParams()` and passing `data: {}` allowed me to customize the preview edge style.

**Comment by umdstu** [comment]
Just circling back on this, it seems I was incorrect about the edgeParams function to be able to effect the classes of the final or ghost edges. It let's you set 'data' classes, that are used by the cytoscape-sbgn-stylesheet, but not the actual edges classes for stylesheet purposes.

TLDR;
It would still be useful to be able to effect the style of ghost and drawn edges created from this plugin via classes.

**Comment by maxkfranz** [comment]
It would make sense to be able to set additional classes.  The base classes
set by the extension should not be overridden.  Are you interested in
making a PR for this feature?

**Comment by umdstu** [comment]
Absolutely, i'll work on it tonight. Thanks @maxkfranz !
